### PR TITLE
Improve flow support

### PR DESCRIPTION
**Pull request by Komalov**
### What is this
It has 150+ errors related to flow right now. Some of them can be fixed without any noticeable changes so it can be suppressed to just 28 errors easily

### CODE REVIEW

**Comment by aaronabramov** [comment]
this is great! thanks for working on this!

how hard would it be to fix those 28 errors? if it's OSS/GH specific, can we add a bunch of $FlowFixMe's and add `yarn flow` to the list of steps on CI?

**Comment by aaronabramov** [comment]
actually nevermind. `yarn flow` command is already there and broken, so this is definitely improving the situation already. let me import it and see if it works with internal flow build as well

**Comment by Komalov** [comment]
It requires more deep understanding of how things work what I currently do not have. But it looks like some stuff more related to flow itself e.g. https://github.com/facebookexperimental/Recoil/blob/c336325c76a68ace6e8da53b28866e442613d855/src/util/Recoil_mergeMaps.js#L22 . Seems flow doesn't notice assignment into the loop condition

File: src/adt/Recoil_Loadable.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ type ResolvedLoadablePromiseInfo<+T> = $ReadOnly<{
 
 export type LoadablePromise<T> = Promise<ResolvedLoadablePromiseInfo<T>>;
 
-type Accessors<T> = $ReadOnly<{
+type Accessors<T> = $ReadOnly<{|
```
Comment (aaronabramov) [inline src/adt/Recoil_Loadable.js:35]: seems like objects are now exact by default, is it not the case in this configuration?

Comment (Komalov) [reply src/adt/Recoil_Loadable.js:35]: I didn't know that, but sure enough I needed to make exact types here manually. Need to check if [utility type](https://flow.org/en/docs/types/utilities/#toc-readonly) behaviours differently

**Pull request closed without merging**